Products/ZenModel/ZenPack.py

```diff
@@ -1201,67 +1202,54 @@ def getServiceTag(self):
         return self.moduleName()
 
 
-    def getServiceDefinitionFiles(self):
+    def hasServiceDefinitions(self):
         """
-        Returns a list of files containing services to be installed.
-
-        This routine can be overridden in order to supply a non-standard list of
-         files. See installServicesFromFiles() for file format.
-        :returns: absolute file paths
-        :rtype: list of strings
+        Determines whether or not this ZenPack has service definitions
+        available for installation.
         """
-        return glob.glob(self.path('service_definition', '*.json'))
+        return any([self.isServicePath(p) for p in glob.glob(self.path('service_definition', '*'))])
 
 
-    def installServices(self):
+    def installServices(self, services=None):
         """
-        Install ControlPlane services for this ZenPack
+        Install ControlCenter services for this ZenPack
         @return: None
         """
         if not ZenPack.currentServiceId:
             return
-        if self.getServiceDefinitionFiles():
-            sdFiles = self.getServiceDefinitionFiles()
-            toConfigPath = lambda x: os.path.join(os.path.dirname(x),'-CONFIGS-')
-            configFileMaps = [DirectoryConfigContents(toConfigPath(i)) for i in sdFiles]
-            self.installServicesFromFiles(sdFiles, configFileMaps, self.getServiceTag())
-        elif self.getDaemonNames():
-            templateLocation = zenPath('Products/ZenModel/data/default_service.json')
-            template = open(templateLocation, 'r').read()
-            daemonPaths = glob.glob(os.path.join(self.getDaemonPath(), '*'))
-            self.installDefaultCollectorServices(daemonPaths,
-                                                 template,
-                                                 self.getServiceTag())
+        if services is None:
+            if self.hasServiceDefinitions():
+                services = self.getServiceDefinitionsFromFiles()
+            elif self.getDaemonNames():
+                services = self.getDefaultCollectorServiceDefinitions()
+        cpClient = ControlPlaneClient(**getConnectionSettings())
+        serviceTree = ServiceTree(cpClient.queryServices("*"))
+        ctx = servicemigration.ServiceContext()
+        for service in services:
+            parentServices = serviceTree.matchServicePath(ZenPack.currentServiceId, service["servicePath"])
+            for parentService in parentServices:
+                ctx._ServiceContext__deployService(json.dumps(service["serviceDefinition"]), parentService.id)
```

Comment:
Oh yes, we're counting on it. Our plan is to support the official interface, and people can of course muck about with the python to change whatever they want, but we can't support it.

The issue is that ZenPacks deploy their service definitions by referencing the ID of the parent service. Since we don't expose the parent ID in the service migration API, it doesn't make sense to expose a function that uses it.